### PR TITLE
Add additional info about deploy and user keys

### PR DESCRIPTION
**Pull request by teesloane**
- updated information about deploy and user keys.
- These docs are ported directly from `https://circleci.com/gh/<USER>/<PROJECT>/edit#checkout`
- Made tone changes / fixed some typos. Will probably want to circle around and make changes in the UI itself.

### CODE REVIEW

File: jekyll/_cci2/gh-bb-integration.md [hunk 1]
```diff
@@ -286,5 +291,24 @@ jobs:
...
+- automatic git merging/tagging/etc.
+
+That's why a deploy key isn't sufficiently powerful for projects with additional private dependencies!
+
+### What about security?
+
+The private keys of the checkout keypairs we generate never leave our systems
+(only the public key is transmitted to GitHub) and are safely encrypted in storage. However, since they are installed into your build containers, any code that you run in CircleCI can read them. You shouldn't push untrusted code to CircleCI!
+
+**Isn't there a middle ground between deploy keys and user keys?**
```
Comment (michelle-luna) [inline jekyll/_cci2/gh-bb-integration.md]: change to Differences (middle ground is folksy)

File: jekyll/_cci2/gh-bb-integration.md [hunk 2]
```diff
@@ -286,5 +291,24 @@ jobs:
...
+
+Not really (unfortunately!). Deploy keys and user keys are the only key types that GitHub supports. Deploy keys are globally unique (i.e. there's no way to make a deploy key with access to multiple repositories) and user keys have no notion of _scope_ separate from the user they're associated with.
```
Comment (michelle-luna) [inline jekyll/_cci2/gh-bb-integration.md]: Remove first sentence fragment
i.e.-> for example
there's-> there is or 'no mechanism exists to'
they're-> user associated with them

**Comment by teesloane** [comment]
Changes made!

File: jekyll/_cci2/gh-bb-integration.md [hunk 3]
```diff
@@ -196,14 +196,19 @@ The account and permissions system we use is not as clear as we would like and a
...
+VCS (GitHub or Bitbucket) for your project. A deploy key is a repo-specific SSH
+key. If you are using Github as your VCS then GitHub has the public key, and
+CircleCI  stores the private key. The deployment key gives CircleCI access to a single repository.
+To prevent CircleCI from pushing to your repository, this deployment key is read-only.
+
+If you want to push to the repository from your builds, you will need a deployment key with write access (user key). The steps to create a user key depend on your VCS.
+
+**What is a user key?**
+
+A user key is a user-specific SSH key. Your VCS has the public key, and we store the private key. Possession of the private key gives the ability to act as that user, for purposes of 'git' access to projects.
```
Comment (michelle-luna) [inline jekyll/_cci2/gh-bb-integration.md]: we -> CircleCI

File: jekyll/_cci2/gh-bb-integration.md [hunk 4]
```diff
@@ -286,5 +291,26 @@ jobs:
             - "SO:ME:FIN:G:ER:PR:IN:T"
 ```
 
-When you push to your Bitbucket project from a job,
-CircleCI will use the SSH key you added.
+### How are these keys used?
+
+When CircleCI build your project, we install the private key into the .ssh directory, and configure SSH to use it when communicating with your version control provider. Therefore, it gets used for:
```
Comment (michelle-luna) [inline jekyll/_cci2/gh-bb-integration.md]: build -> builds
we-> the private key is installed...and SSH is configured...
.ssh -> `.ssh`

File: jekyll/_cci2/gh-bb-integration.md [hunk 5]
```diff
@@ -286,5 +291,26 @@ jobs:
...
+
+- checking out the main project
+- checking out any GitHub-hosted submodules
+- checking out any GitHub-hosted private dependencies
+- automatic git merging/tagging/etc.
+
+For this reason, a deploy key isn't sufficiently powerful for projects with additional private dependencies!
```
Comment (michelle-luna) [inline jekyll/_cci2/gh-bb-integration.md]: remove !

File: jekyll/_cci2/gh-bb-integration.md [hunk 6]
```diff
@@ -286,5 +291,26 @@ jobs:
...
+
+### What about security?
+
+The private keys of the checkout keypairs CircleCI generates never leave the CircleCI systems (only the public key is transmitted to GitHub) and are safely encrypted in storage. However, since they are installed into your build containers, any code that you run in CircleCI can read them.
+
+**Isn't there a Difference between deploy keys and user keys?**
```
Comment (michelle-luna) [inline jekyll/_cci2/gh-bb-integration.md]: no caps on Difference in the middle of the sentence

**Pull request merged**